src/components/register/RegisterSection.tsx

```diff
@@ -0,0 +1,190 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import GoogleIcon from '../../assets/googleIcon.svg';
+import Button from '../Button';
+import Input from '../Input';
+import Select from '../Select';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { signUp } from '../../state/register/registerSlice';
+
+type Formfields = z.infer<typeof extendedSchema>;
+interface ExtendedFormFields extends Formfields {
+  confirmPassword: string;
+}
+const dbLinks = {
+  local: 'http://localhost:8080/api/users/signup',
+  hosted: 'https://e-commerce-mavericcks-bn-staging-istf.onrender.com/api/users/signup',
+};
+const RegisterSection = () => {
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+
+  const {
+    register,
+    handleSubmit,
+    setError,
+    formState: { errors, isSubmitting },
+  } = useForm<ExtendedFormFields>({
+    resolver: zodResolver(extendedSchema),
+  });
+  const onSubmit: SubmitHandler<ExtendedFormFields> = async retrievedData => {
+    const { password, lastName, firstName, gender, phoneNumber, email } = retrievedData;
+    try {
+      const response: Response = await fetch(dbLinks['hosted'], {
```

Comment:
can you use Redux ToolKit Query instead of this. I think it is what we agreed on using